### ISSUE TITLE
nvals weirdness with 8.0.0 for max-sized Vector

### ISSUE THREAD

**Issue by eriknw** [DrTimothyAldenDavis/GraphBLAS#215]
```C
  GrB_Vector_new(&v_1, GrB_INT64, 1152921504606846976);
  GrB_Vector_assign_INT64(v_1, NULL, NULL, 1, GrB_ALL, 1152921504606846976, NULL);
  GrB_Vector_nvals(&s_nvals, v_1);  // 9223372036854775807, but should be 1152921504606846976
```
Note that `2^60 == 1152921504606846976` and `2^63 - 1 == 9223372036854775807`. I see the same results with Matrix with one of the dimensions of length `2^60`.

I get correct nvals for smaller Vectors, including Vector of size `2^60 - 1`.

**Comment by DrTimothyAldenDavis** [comment]
I return UINT64_MAX if the product of nrows times ncols can cause int64_t overflow, but I might be overly cautious in this case because ncols is 1 for a GrB_Vector.

It would be a good idea to use 1152921504606846976L to make sure that value is a 64-bit constant.

**Comment by eriknw** [comment]
Gotcha. I actually cause this error from Python via `python-graphblas` (where types are handled correctly):
```python
v = Vector(int, size=2**60)
v[:] = 1
v.nvals  # bad value!
```
This worked correctly with previous versions of SuiteSparse:GraphBLAS. v8.0.0 introduced a regression.

**Comment by DrTimothyAldenDavis** [comment]
Try the latest push to the master branch.  I changed how I handle checking integer overflow in Source/GB_uint64_multiply.h in v8.0.0 and I think I introduced this bug then.  The latest push to the master branch should fix this.

**Comment by DrTimothyAldenDavis** [comment]
That fixed it.  I added a new test (GraphBLAS/Test/GB_mex_test25.c) that fails with v8.0.0 (triggering this error) but works fine for the draft v8.0.1.

**Comment by DrTimothyAldenDavis** [comment]
The fix isn't perfect because the integer overflow guard is not exact (it wasn't exact in v7.x either).  For example, if you ask for GrB_Matrix_nvals for a matrix of size 2 by 2^30, it will report INT64_MAX, or 'infinity' because that case triggers the overflow guard.  That's by design.  It's too costly to put in a precise integer overflow guard.

**Comment by eriknw** [comment]
I confirm that this is now fixed in v8.0.1. Thanks @DrTimothyAldenDavis! Closing.
